mesh/mesh.hpp

```diff
@@ -579,8 +579,27 @@ class Mesh
        new mesh. Periodic meshes are not supported. */
    static Mesh MakeSimplicial(const Mesh &orig_mesh);
 
+   /// Create a periodic mesh by identifying vertices of @a orig_mesh.
+   /** Each vertex @a i will be mapped to vertex @a v2v[i], such that all
+       vertices that are coincident under the periodic mapping get mapped to
+       the same index. The mapping @a v2v can be generated from translation
+       vectors using Mesh::CreatePeriodicVertexMapping. */
+   static Mesh MakePeriodic(const Mesh &orig_mesh, const std::vector<int> &v2v);
+
    ///@}
 
+   /// @brief Creates a mapping @a v2v from the vertex indices of the mesh such
+   /// that coincident vertices under the given @a translations are identified.
+   /** Each Vector in @a translations should be of size @a sdim (the spatial
+       dimension of the mesh). Two vertices are considered coincident if the
+       translated coordinates of one vertex are within the given tolerance (@a
+       tol, relative to the mesh diameter) of the coordinates of the other
+       vertex.
+       @warning This algorithm does not scale well with the number of vertices
```

Comment:
cc: @mlstowell, do you suggest a change of the wording in this comment?

Reply:
It may be best to say "This algorithm does not scale well with the number of boundary vertices". This is more precise and will allow the user to better gauge whether or not this is appropriate for their meshes.

Reply:
Another important warning we should include involves the number of edges separating opposite boundary surfaces. I'm not sure how best to word this but MFEM periodic meshes require no fewer than three edges connecting periodic boundaries which ensures that each edge can be identified by a unique pair of vertices.

Reply:
Yes, but it's more than that. Three vertices forming two connected edges e.g. `a--b--c`, are also not allowed because identifying the first and last vertices of the triplet will produce two edges, `a--b` and `b--a`, which cannot be distinguished by the vertices alone.

Reply:
I added a comment to that effect in `Mesh::MakePeriodic`.